Auto blacklist erroring entities

meta1203 opened the pull request:
Just like the title says: if an entity causes an exception in the default SerDes filter, automatically add it to the blacklist

This also adds the blacklisted entity to its own config file to persist between sessions.

(It ALSO fixes a bug related to nightconfig's TOML implementation & Java's internal Map object)

jediminer543 commented on src/main/java/org/jmt/mcmt/serdes/filter/AutoFilter.java:
```diff
@@ -0,0 +1,48 @@
+
+	@Override
+	public ClassMode getModeOnline(Class<?> c) {
+		return ClassMode.BLACKLIST;
```
This will cause every unknown class to be blacklisted and run through this filter, which may not be intended behaviour, and definately isn't really desirable; this should return `ClassMode.UNKNOWN` as it's not got any of it's own filtering logic.

jediminer543 commented:
You work fast

I'm curently reading through it now but most of it looks okay, I've added a comment to one of the things that's a bit broken, could you say something about it being licenced under BSD-2-Clause License in a comment just so I don't need to worry about wierd legal shenanagins (as IDK how well infering it from your repo works legally and I've seen bad things happen).

I've added a comment about a thing that probs needs changing; I'll try double checking everything through more again just to check. Once that's fixed and I'm not worrying about legal things I can merge it.

meta1203 commented:
@jediminer543 Went ahead and fixed that bug, thanks for catching that. I also added that license acknowledgement for you (lemme know if you need me to put it elsewhere). I wouldn't ever try to undermine an OSS project, but after what happened with CraftBukkit I don't blame you for being cautious.
